bee-message/src/output/mod.rs

```diff
@@ -201,10 +204,71 @@ impl Output {
             Self::Nft(output) => Some(output.immutable_feature_blocks()),
         }
     }
+
+    /// Checks if a sufficient storage deposit was made for the given [`Output`].
+    pub fn check_sufficient_storage_deposit(&self, config: &ByteCostConfig) -> Result<(), Error> {
```

Comment:
I would call this simply `check_storage_deposit` as given the errors it checks a bit more than just sufficiency. For a bit more consistency with other verification methods I would even call it `verify_storage_deposit`.